### ISSUE TITLE
Fullscreen runs on second monitor and closes the main one in dual monitor setup, Linux

### ISSUE THREAD

**Issue by naezith** [SFML/SFML#1226]
I can't test this myself but a full-time Linux user reported it,

He has two monitors, same model, both **1920x1080**. He opens the game in full-screen, at first launch game picks the **sf::VideoMode::getDesktopMode()** -> **3840x1080** which is both monitors combined. It's not terrible but still is every user has to change it so I consider it as a bad thing. When he changes it to **1920x1080**, game runs on the second monitor only, and **main monitor shuts down**, no signal.

Can someone with similar setup try this?

**Comment by MarioLiebisch** [comment]
It's a known issue with multi monitor setups. This is essentially an extension of issue #188.

**Comment by eXpl0it3r** [comment]
This sounds more like a xrandr (or whatever these Linux display settings things are called) issue, where they defined just one monitor by joining them together. By reducing the area, it will just start from the left side potentially ending up on the second monitor.

Without actually someone that can provide proper information on their setup and has the Linux knowledge required to understand how these monitor things work, there's little we can do at this point.

This issue also reminded me of #436, but we've switched back to Xlib in the mean time.

**Comment by Yengas** [comment]
I had a problem when i tried to play @naezith's game. I couldn't change the resolution for borderless, windowed or fullscreen modes. After trying out different display configurations in my Desktop Environment, we tried to see what was SFML returning when Naezith's game called **getFullscreenModes**. We compiled SFML from the source code and ran a [sample code](https://gist.github.com/Yengas/ab23e1e68f30c01fd8b0cdf911899015). The sample application prints out the fullscreen modes returned by the SFML lib. And it prints out my current desktop resolution 7 times, with the exact same width and height.

So if i disable my second monitor and run the sample application i get this;
```
Total size of fullscreen modes: 7
1920x1080 # x7 times
```
With the dual monitor;
```
Total size of fullscreen modes: 7
3840x1080 # x7 times
```

SFML Version: **git c43b5991afe815ef92a812d67f8af12b0143adfd**
G++: **g++ (Ubuntu 5.4.1-2ubuntu1~16.04) 5.4.1 20160904**
Linux: **Linux berserk 4.4.0-77-generic #98-Ubuntu SMP Wed Apr 26 08:34:02 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux**
Linux Distro: **Ubuntu 16.04.2 LTS**

**Comment by bloodsword** [comment]
A user of my game reported it. 
I reproduced the problem. The most annoying thing is that the monitor stays off after the application closed and windows are rearranged.

After some investigation, I found that XRRSetCrtcConfig should be used instead of XRRSetScreenConfig.

I gave it a try (https://github.com/bloodsword/SFML/tree/dual-monitor-fix).
It's not shutting down the screen anymore, but some WM (Cinnamon and Unity at least) do not switch the window to fullscreen when there are 2 monitors. 
Ubuntu WM was already suffering this problem without the fix, I guess it's the issue #1224. 
I have a fix for this issue but it's quite ugly, I will try to post it soon.

To use the mouse in the second screen, you need to alt+tab. 
This behavior suits my game, since screen edges are used to move the camera.
But I don't really understand why.

Here are some useful links :
https://www.reddit.com/r/gamedev/comments/2w8ahe/changing_resolutions_for_fullscreen_in_x11_linux
https://sourcecodebrowser.com/libxrandr/1.2.3/_xrandr_8h.html

**Comment by eXpl0it3r** [comment]
@bloodsword Feel free to open a pull request. Makes it easier to review and allows our CI system to build it.

**Comment by barthoukes** [comment]
Thanks, I get the impression that it may run on a Linux POS system PP1455 with an additional customer display. The main touchscreen is 1024x768 and the customer display (not touch) is 1024x600 pixels. Example pictures at www.kassaku.nl. We try to move our application from SDL to SFML. Are there any other pitfalls we should know about?

**Comment by eXpl0it3r** [comment]
Merged #1238 in 46ce05cd9c336020e911a1f6a3486132dc31b679

**Comment by SudoTheHacker** [comment]
Hello, the fix can be added to the official version of SFML, because I don't want to search and compile every time.

**Comment by eXpl0it3r** [comment]
It's available already, fixed as part of SFML 2.5.1
